src/neighborhoods/ellipsoid.jl

```diff
@@ -0,0 +1,46 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    EllipsoidNeighborhood(semiaxes, angles; convention=:TaitBryanExtr)
+
+An ellipsoid neighborhood with `semiaxes` and `angles`. For 2D ellipses,
+there are two semiaxes and one rotation angle. For 3D ellipsoids, there are
+three semiaxes and three rotation angles. Different rotation conventions can be
+passed via the `convention` keyword argument. The list of conventions is
+available in the [aniso2distance](@ref) documentation.
+"""
+struct EllipsoidNeighborhood{S,A,C,M} <: AbstractBallNeighborhood
+  semiaxes::S
+  angles::A
+  convention::C
+  metric::M
+
+  function EllipsoidNeighborhood{S,A,C,M}(semiaxes, angles; convention) where {S,A,C,M}
+    metric = aniso2distance(semiaxes, angles, convention=convention)
+    new(semiaxes, angles, convention, metric)
+  end
+end
+
+EllipsoidNeighborhood(semiaxes::S, angles::A; convention::C=:TaitBryanExtr) where {S,A,C} =
+  EllipsoidNeighborhood{S,A,C,Mahalanobis}(semiaxes, angles; convention)
```

Comment:
```suggestion
  function EllipsoidNeighborhood{S,A,C,M}(semiaxes, angles convention) where {S,A,C,M}
    metric = aniso2distance(semiaxes, angles, convention=convention)
    new(semiaxes, angles, convention, metric)
  end
end

EllipsoidNeighborhood(semiaxes::S, angles::A; convention::C=:TaitBryanExtr) where {S,A,C} =
  EllipsoidNeighborhood{S,A,C,Mahalanobis}(semiaxes, angles, convention)
```

Reply:
Did you have a chance to commit this suggestion @rmcaixeta ?

Reply:
Haven't noted that one. Now it's adjusted